### PR TITLE
Update to nikic/php-parser 5.x

### PR DESCRIPTION
**Pull request by staabm**
closes https://github.com/TomasVotruba/class-leak/issues/45

with this PR the min PHP version is raised from 7.2 to 7.4, because thats what niki/php-parser requires (maybe it should be a new major version, in case you still need PHP 7.2 support)

### CODE REVIEW

File: src/DependencyInjection/ContainerFactory.php [hunk 1]
```diff
@@ -28,7 +28,7 @@ public function create(): Container
 
         $container->singleton(Parser::class, static function (): Parser {
             $parserFactory = new ParserFactory();
-            return $parserFactory->create(ParserFactory::PREFER_PHP7);
+            return $parserFactory->createForNewestSupportedVersion();
```
Comment (staabm) [inline src/DependencyInjection/ContainerFactory.php:31]: per upgrade guide, this should be roughly equivalent

https://github.com/nikic/PHP-Parser/blob/master/UPGRADE-5.0.md#changes-to-the-parser-factory

**Comment by TomasVotruba** [comment]
Just applied the cs on main. Could you rebase this one?

**Comment by staabm** [comment]
> Could you rebase this one?

done

File: .github/workflows/downgraded_release.yaml [hunk 2]
```diff
@@ -33,16 +33,16 @@ jobs:
             -   run: mkdir rector-local
             -   run: composer require rector/rector --working-dir rector-local --ansi
 
-            # downgrade to PHP 7.2
-            -   run: rector-local/vendor/bin/rector process src bin vendor --config build/rector-downgrade-php-72.php --ansi
+            # downgrade to PHP 7.4
+            -   run: rector-local/vendor/bin/rector process src bin vendor --config build/rector-downgrade-php-74.php --ansi
```
Comment (TomasVotruba) [inline .github/workflows/downgraded_release.yaml:37]: This would make this package useless for upgrade. I've got few PHP 7.2 around. Please revert this change

Comment (staabm) [reply .github/workflows/downgraded_release.yaml]: lets see whether nikic/php-parser will work for 7.2 after it got downgraded

**Comment by TomasVotruba** [comment]
Let's see how it goes :) Thanks :+1:

**Comment by staabm** [comment]
running 1.0.1 on phpunit now leads to

```
mstaab@NB-COMPLEX-61 MINGW64 /c/dvl/Workspace/phpunit (main)
$ /c/tools/php83/php vendor/bin/class-leak check src/ tests/
PHP Fatal error:  Declaration of ClassLeak202410\PhpParser\Lexer\TokenEmulator\MatchTokenEmulator::getPhpVersion(): string must be compatible with ClassLeak202410\PhpParser\Lexer\TokenEmulator\TokenEmulator::getPhpVersion(): ClassLeak202410\PhpParser\PhpVersion in C:\dvl\Workspace\phpunit\vendor\tomasvotruba\class-leak\vendor\nikic\php-parser\lib\PhpParser\Lexer\TokenEmulator\MatchTokenEmulator.php on line 9

Fatal error: Declaration of ClassLeak202410\PhpParser\Lexer\TokenEmulator\MatchTokenEmulator::getPhpVersion(): string must be compatible with ClassLeak202410\PhpParser\Lexer\TokenEmulator\TokenEmulator::getPhpVersion(): ClassLeak202410\PhpParser\PhpVersion in C:\dvl\Workspace\phpunit\vendor\tomasvotruba\class-leak\vendor\nikic\php-parser\lib\PhpParser\Lexer\TokenEmulator\MatchTokenEmulator.php on line 9
```

**Comment by staabm** [comment]
I guess the problem might be that rector itself ships with php-parser 4.x and this is somehow mixed up the class-leak php-parser version on downgrade

**Comment by TomasVotruba** [comment]
Let's revert for now https://github.com/TomasVotruba/class-leak/pull/48
